#!/usr/bin/env python

PACKAGE = "your_package_name"  # Sostituisci con il nome reale del tuo pacchetto

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()



gen.add("time_between_controllers", double_t, 0,
        "Delay between controller switches [s]", 1.0, 0.0, 5.0)

gen.add("object_point_survive_time", double_t, 0,
        "Time a detected object point is kept alive [s]", 15.0, 1.0, 60.0)

gen.add("Ts", double_t, 0,
        "Sampling time [s]", 0.1, 0.01, 1.0)

gen.add("distance_to_goal", double_t, 0,
        "Distance threshold to consider goal reached [m]", 0.5, 0.05, 5.0)

gen.add("frame_id_mpc", str_t, 0,
        "Reference frame used for MPC", "odom")

exit(gen.generate(PACKAGE, "mpc_namespace", "mpcParams"))
